File: arrays_and_strings/compression.py


def compress(text):
    text += " "
    repeat = text[0]
    count = 0
    output = ""
    i = 0

    while i < len(text):
        if text[i] != repeat:
            if count == 1:
                output += str(repeat)
            else:
                output += str(count) + str(repeat)
            
            repeat = text[i]
            count = 0
        else:
            count += 1
            i += 1

    return output

text = "aaaabbbbbaaaadcfbbbbaassssssrrrrrtttvtvt"
print(compress(text))

File: arrays_and_strings/compression.php
<?php

function compression($text){
    $text .= " ";
    $i = 0;
    $repeat = $text[0];
    $count = 0;
    $output = "";

    while ($i < strlen($text)) {
        if($repeat != $text[$i]) {
            $output .= $count.$repeat;
            $repeat = $text[$i];
            $count = 1;
        } else {
            $count += 1;
        }
        $i += 1;
    }

    return $output;
}

$result = compression("aaaabbbbbaaaadcfbbbbaassssssrrrrrtttvtvt");

echo $result;